homeassistant/components/iqvia/__init__.py

```diff
@@ -143,42 +141,100 @@
 class IQVIAData:
     """Define a data object to retrieve info from IQVIA."""
 
-    def __init__(self, client):
+    def __init__(self, hass, client):
         """Initialize."""
+        self._async_cancel_time_interval_listener = None
         self._client = client
+        self._hass = hass
         self.data = {}
         self.zip_code = client.zip_code
 
-        self.fetchers = Registry()
-        self.fetchers.register(TYPE_ALLERGY_FORECAST)(self._client.allergens.extended)
-        self.fetchers.register(TYPE_ALLERGY_OUTLOOK)(self._client.allergens.outlook)
-        self.fetchers.register(TYPE_ALLERGY_INDEX)(self._client.allergens.current)
-        self.fetchers.register(TYPE_ASTHMA_FORECAST)(self._client.asthma.extended)
-        self.fetchers.register(TYPE_ASTHMA_INDEX)(self._client.asthma.current)
-        self.fetchers.register(TYPE_DISEASE_FORECAST)(self._client.disease.extended)
-        self.fetchers.register(TYPE_DISEASE_INDEX)(self._client.disease.current)
+        # Define a registry to hold onto API coroutines:
+        self.api_coros = Registry()
+        self.api_coros.register(TYPE_ALLERGY_FORECAST)(client.allergens.extended)
+        self.api_coros.register(TYPE_ALLERGY_INDEX)(client.allergens.current)
+        self.api_coros.register(TYPE_ALLERGY_OUTLOOK)(client.allergens.outlook)
+        self.api_coros.register(TYPE_ASTHMA_FORECAST)(client.asthma.extended)
+        self.api_coros.register(TYPE_ASTHMA_INDEX)(client.asthma.current)
+        self.api_coros.register(TYPE_DISEASE_FORECAST)(client.disease.extended)
+        self.api_coros.register(TYPE_DISEASE_INDEX)(client.disease.current)
+
+        self._api_category_count = {
+            TYPE_ALLERGY_FORECAST: 0,
+            TYPE_ALLERGY_INDEX: 0,
+            TYPE_ALLERGY_OUTLOOK: 0,
+            TYPE_ASTHMA_FORECAST: 0,
+            TYPE_ASTHMA_INDEX: 0,
+            TYPE_DISEASE_FORECAST: 0,
+            TYPE_DISEASE_INDEX: 0,
+        }
+        self._api_category_locks = {
+            TYPE_ALLERGY_FORECAST: asyncio.Lock(),
+            TYPE_ALLERGY_INDEX: asyncio.Lock(),
+            TYPE_ALLERGY_OUTLOOK: asyncio.Lock(),
+            TYPE_ASTHMA_FORECAST: asyncio.Lock(),
+            TYPE_ASTHMA_INDEX: asyncio.Lock(),
+            TYPE_DISEASE_FORECAST: asyncio.Lock(),
+            TYPE_DISEASE_INDEX: asyncio.Lock(),
+        }
+
+    async def _async_get_data_from_api(self, api_category):
+        """Update and save data for a particular API category."""
+        if self._api_category_count[api_category] == 0:
+            return
+
+        try:
+            self.data[api_category] = await self.api_coros[api_category]()
+        except IQVIAError as err:
+            _LOGGER.error("Unable to get %s data: %s", api_category, err)
+            self.data[api_category] = {}
```

Comment:
You only check if it's truthy in your entities, so `None` should be enough
```suggestion
            self.data[api_category] = None
```